core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/HealthCheckContextGroup.java

```diff
@@ -54,16 +55,45 @@ List<Endpoint> candidates() {
     }
 
     void initialize() {
-        initFutures = CompletableFuture.allOf(contexts.values().stream()
-                                                      .peek(context -> {
-                                                          if (!context.isInitialized()) {
-                                                              // A newly created context
-                                                              context.init(checkerFactory.apply(context));
-                                                          }
-                                                      })
-                                                      .map(DefaultHealthCheckerContext::whenInitialized)
-                                                      .collect(toImmutableList())
-                                                      .toArray(EMPTY_FUTURES));
+        final List<CompletableFuture<Void>> futures =
+                contexts.values().stream()
+                        .peek(context -> {
+                            if (!context.isInitialized()) {
+                                // A newly created context
+                                context.init(checkerFactory.apply(context));
+                            }
+                        })
+                        .map(DefaultHealthCheckerContext::whenInitialized)
+                        .collect(toImmutableList());
+
+        initFutures = CompletableFuture.allOf(futures.toArray(EMPTY_FUTURES)).handle((unused, cause) -> {
+            if (cause == null) {
+                return null;
+            }
+            if (futures.isEmpty()) {
+                return null;
+            }
+
+            if (futures.stream().anyMatch(future -> !future.isCompletedExceptionally())) {
+                // There is at least one success
+                return null;
+            }
+
+            Throwable combined = null;
+            for (CompletableFuture<Void> future : futures) {
+                try {
+                    future.join();
+                } catch (Throwable ex) {
+                    if (combined == null) {
+                        combined = ex;
+                    } else {
+                        combined.addSuppressed(Exceptions.peel(ex));
```

Comment:
optional nit; I'm wondering if there's no need to also truncate the number of suppressed exceptions 🤔    I'm imagining if there are many endpoints the logged stacktrace may be quite long (although I guess it is only printed once so it might not matter)

Reply:
As you said, it is printed once and it might happen in an unusual/rare case. So I don't see we need to truncate it.